src/backend/input.rs

```diff
@@ -216,6 +231,9 @@ pub trait InputBackend: Sized {
     /// Get current `InputConfig`
     fn input_config(&mut self) -> &mut Self::InputConfig;
 
+    /// Called to inform the Input backend about a new focused Output for a `Seat`
+    fn set_output_metadata(&mut self, seat: &Seat, output: &Output);
```

Comment:
We could convert them, like I did for the `TouchEvent` already. That makes matching easier and I don't think the current `TouchEvent` is missing something critical. I could also model a backend-agnostic `PointerEvent` based on libinputs event.

I fear to many generics and traits will make it look much less platform-independently then it actually is.